design/20230726-cel-policy.md

```diff
@@ -0,0 +1,369 @@
+
+This proposal suggests to add support for specifying allowed string fields using CEL expressions, similar to the existing
+wildcard support. While CEL expressions could eventually replace the current CSR attribute validation API - including the 
+wildcard support, the main improvement would be support for policies that specifies an allowed attribute value that depends
+on CertificateRequest fields e.g. `namespace`. If implemented, this new feature would close
+[#62](https://github.com/cert-manager/approver-policy/issues/62).
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this proposal gets implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### X.509 DNS SAN in namespace domain
+
+As a cluster admin in a multi-tenant cluster, I want to provide a cert-manager ClusterIssuer to our users allowing
+self-provisioning of server certificates. Since the cluster is multi-tenant, I want to ensure that tenants do not
+"trip on each other's toes" and enforce that the issuer only signs certificates where the X.509 DNS SAN attribute
+ends with `<namespace>.<domain>`.
+
+A variant of this user story is exemplified in [#62](https://github.com/cert-manager/approver-policy/issues/62),
+and I think it's one of the most common user stories the proposed feature will solve.
+
+#### SPIFFE ID identifying namespace
+
+As a cluster admin in a multi-tenant cluster, I want to provide a cert-manager ClusterIssuer to our users allowing
+self-provisioning of [X.509 SVID certificates](https://github.com/spiffe/spiffe/blob/main/standards/X509-SVID.md).
+The [SPIFFE ID](https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE-ID.md) (X.509 URI SAN) should identify
+the namespace (and service account), and must start with `spiffe://<trust-domain>/ns/<namespace>/sa/`.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go into as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes/PKI ecosystem.
+
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+An [experimental approver-policy plugin](https://github.com/erikgb/cel-approver-policy-plugin) providing some of the
+requested features exists, and it's proposed to base the design on this plugin.
```

Comment:
this line is awkward. https://en.wikipedia.org/wiki/Garden-path_sentence

In general, I'd discourage the use of abbreviations (`'`), 
```suggestion
requested features exists, and we propose to base the design on this plugin.
```